catalog/errors.go

```diff
@@ -28,4 +28,7 @@ var (
 	ErrReadEntryTimeout              = errors.New("read entry timeout")
 	ErrMissingDiffResultsIDInContext = errors.New("missing diff results id in context")
 	ErrInvalidValue                  = errors.New("invalid value")
+	ErrNonDirectNotSupported         = errors.New("non direct diff not supported")
+	ErrNonDirectMergeNotSupported    = errors.New("merge suported only between branches that are parent-child or child-parent")
+	ErrSameBranchMergeNotSupported   = errors.New("same branch merge not supported")
```

Comment:
1. Use ErrFeatureNotSupported (%w) when you build these errors as all re from this type
2. typo: supported
3. ErrNonDirectNotSupported - missing "branch"